File: server_integration_tests/java/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/CollideServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.CollideService;
import dev.graphlink.javaserver.generated.types.Collide;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
public class CollideServiceImpl implements CollideService {

    @Override
    public Mono<Collide> collide() {
        // The two colliding wire fields (`class` and `class_`) carry distinct
        // values so the round-trip can prove they stayed separate identifiers.
        return Mono.just(Collide.builder().id("1").class_2("A").class2("B").build());
    }
}

File: server_integration_tests/kotlin/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/CollideServiceImpl.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.services.CollideService
import dev.graphlink.kotlinserversuspend.generated.types.Collide
import org.springframework.stereotype.Service

@Service
class CollideServiceImpl : CollideService {
    // The two colliding wire fields (`class` and `class_`) carry distinct
    // values so the round-trip can prove they stayed separate identifiers.
    override suspend fun collide(): Collide = Collide(id = "1", class_2 = "A", class2 = "B")
}
